# **TF-IDF**

Indicará o quão raro uma palavra é, levando em consideração a sua frequência dentro da frase, quando menos uma palavra aparece mais raro ela é.

In [1]:
import pandas as pd
import numpy as np

from sklearn.feature_extraction.text import TfidfVectorizer


text = [
    "o amigo do meu primo esta faculdade",
    "a faculdade e perto",
    "meu amigo é mais velho",
    "saber matematica e importante",
    "inglês é um idioma muito falado"
]


In [2]:
vect = TfidfVectorizer()
vect.fit(text)

TfidfVectorizer()

In [3]:
vect.vocabulary_

{'amigo': 0,
 'do': 1,
 'meu': 10,
 'primo': 13,
 'esta': 2,
 'faculdade': 3,
 'perto': 12,
 'mais': 8,
 'velho': 16,
 'saber': 14,
 'matematica': 9,
 'importante': 6,
 'inglês': 7,
 'um': 15,
 'idioma': 5,
 'muito': 11,
 'falado': 4}

In [4]:
vect.idf_

array([1.69314718, 2.09861229, 2.09861229, 1.69314718, 2.09861229,
       2.09861229, 2.09861229, 2.09861229, 2.09861229, 2.09861229,
       1.69314718, 2.09861229, 2.09861229, 2.09861229, 2.09861229,
       2.09861229, 2.09861229])

In [8]:
menor_valor = min(vect.idf_)
maior_valor = max(vect.idf_)

print(f"O menor valor equivale a {menor_valor} e o maior valor a {maior_valor}")

O menor valor equivale a 1.6931471805599454 e o maior valor a 2.09861228866811


Ou seja, os itens ***mais raros*** possuem o valor **2.09861228866811** como referência.<br>
E o ***menos raros*** possuem o valor **1.6931471805599454**